File: win-prog/ProgWinVB/Text and Fonts/AllAboutFont/AllAboutFont.vb
'---------------------------------------------
' AllAboutFont.vb (c) 2002 by Charles Petzold
'---------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class AllAboutFont
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New AllAboutFont())
    End Sub

    Sub New()
        Text = "All About Font"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        grfx.DrawString( _
                "Name: " & Font.Name & vbLf & _
                "FontFamily: " & Font.FontFamily.ToString() & vbLf & _
                "FontStyle: " & Font.Style.ToString() & vbLf & _
                "Bold: " & Font.Bold & vbLf & _
                "Italic: " & Font.Italic & vbLf & _
                "Underline: " & Font.Underline & vbLf & _
                "Strikeout: " & Font.Strikeout & vbLf & _
                "Size: " & Font.Size & vbLf & _
                "GraphicsUnit: " & Font.Unit.ToString() & vbLf & _
                "SizeInPoints: " & Font.SizeInPoints & vbLf & _
                "Height: " & Font.Height & vbLf & _
                "GdiCharSet: " & Font.GdiCharSet & vbLf & _
                "GdiVerticalFont: " & Font.GdiVerticalFont & vbLf & _
                "GetHeight(): " & Font.GetHeight() & vbLf & _
                "GetHeight(grfx): " & Font.GetHeight(grfx) & vbLf & _
                "GetHeight(100 DPI): " & Font.GetHeight(100), _
                Font, New SolidBrush(clr), PointF.Empty)
    End Sub
End Class

File: win-prog/ProgWinCS/Text and Fonts/AllAboutFont/AllAboutFont.cs
//-------------------------------------------
// AllAboutFont.cs © 2001 by Charles Petzold
//-------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class AllAboutFont: PrintableForm
{
     public new static void Main()
     {
          Application.Run(new AllAboutFont());
     }
     public AllAboutFont()
     {
          Text = "All About Font";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          grfx.DrawString(
                    "Name: "               + Font.Name            + "\n" +
                    "FontFamily: "         + Font.FontFamily      + "\n" +
                    "FontStyle: "          + Font.Style           + "\n" +
                    "Bold: "               + Font.Bold            + "\n" +
                    "Italic: "             + Font.Italic          + "\n" +
                    "Underline: "          + Font.Underline       + "\n" +
                    "Strikeout: "          + Font.Strikeout       + "\n" +
                    "Size: "               + Font.Size            + "\n" +
                    "GraphicsUnit: "       + Font.Unit            + "\n" +
                    "SizeInPoints: "       + Font.SizeInPoints    + "\n" +
                    "Height: "             + Font.Height          + "\n" +
                    "GdiCharSet: "         + Font.GdiCharSet      + "\n" +
                    "GdiVerticalFont : "   + Font.GdiVerticalFont + "\n" +
                    "GetHeight(): "        + Font.GetHeight()     + "\n" +
                    "GetHeight(grfx): "    + Font.GetHeight(grfx) + "\n" +
                    "GetHeight(100 DPI): " + Font.GetHeight(100),
                    Font, new SolidBrush(clr), Point.Empty);
     }
}
